crates/trident/src/diagnostics.rs

```diff
@@ -0,0 +1,779 @@
+    full_dump: bool,
+) -> Result<PathBuf, Error> {
+    let mut collected_files = Vec::new();
+    let file = osutils::files::create_file(output_path)?;
+    let encoder = zstd::Encoder::new(file, 0)?;
+    let mut tar = tar::Builder::new(encoder);
+
+    for file_to_collect in files_to_collect {
+        if let Ok(meta) = fs::metadata(&file_to_collect.src) {
+            if let Err(e) = tar.append_path_with_name(
+                &file_to_collect.src,
+                format!("{}/{}", DIAGNOSTICS_BUNDLE_PREFIX, file_to_collect.tar_path),
+            ) {
+                warn!(
+                    "Failed to add {} to diagnostics bundle: {}",
+                    file_to_collect.tar_path, e
+                );
+                continue; // Skip this file
+            }
+            collected_files.push(FileMetadata {
+                path: file_to_collect.tar_path,
+                size_bytes: meta.len(),
+                description: file_to_collect.desc,
+            });
+        } else {
+            debug!(
+                "File {} does not exist, skipping",
+                file_to_collect.src.display()
+            );
+        }
+    }
+
+    // Collect and write full journal directly to tarball if requested
+    if full_dump {
+        if let Some(journal_content) = collect_full_journal() {
+            let mut header = tar::Header::new_gnu();
+            header.set_size(journal_content.len() as u64);
+            header.set_mode(0o644);
+            header.set_mtime(
+                std::time::SystemTime::now()
+                    .duration_since(std::time::UNIX_EPOCH)
+                    .unwrap_or_default()
+                    .as_secs(),
+            );
+            header.set_cksum();
+            tar.append_data(
+                &mut header,
+                format!("{}/full-journal", DIAGNOSTICS_BUNDLE_PREFIX),
+                journal_content.as_bytes(),
+            )?;
+            collected_files.push(FileMetadata {
+                path: "full-journal".to_string(),
+                size_bytes: journal_content.len() as u64,
+                description: "Full system journal from current boot".to_string(),
+            });
+        }
+    }
+
+    report.collected_files = Some(collected_files);
+    let report_json = serde_json::to_string_pretty(report)?;
+
+    let mut report_file_header = tar::Header::new_gnu();
+    report_file_header.set_size(report_json.len() as u64);
+    report_file_header.set_mode(0o644);
+    report_file_header.set_mtime(
+        std::time::SystemTime::now()
+            .duration_since(std::time::UNIX_EPOCH)
+            .unwrap_or_default()
+            .as_secs(),
+    );
+    report_file_header.set_cksum();
+    tar.append_data(
+        &mut report_file_header,
+        format!("{}/report.json", DIAGNOSTICS_BUNDLE_PREFIX),
+        report_json.as_bytes(),
+    )?;
+
+    tar.into_inner()?.finish()?;
+
+    Ok(output_path.to_path_buf())
+}
+
+pub(crate) fn generate_and_bundle(
```

Comment:
as a general though, i think we should address errors more. My thinking is that the only fatal error in the diagniostics flow is an error writting the final tarbal to disk, everything else we should log.
The diagnostics bundle should contain all the data was was successfully collected, and some list on what was NOT collected, and why (probably just the body of each error)